samza-api/src/main/java/org/apache/samza/system/SystemAdmin.java

```diff
@@ -84,12 +95,22 @@ default void validateStream(StreamSpec streamSpec) throws StreamValidationExcept
   }
 
   /**
-   * Clear the stream described by the spec.
+   * Clear the entire stream described by the spec.
    * @param streamSpec  The spec for the physical stream on the system.
    * @return {@code true} if the stream was successfully cleared.
    *         {@code false} if clearing stream failed.
    */
   default boolean clearStream(StreamSpec streamSpec) {
     throw new UnsupportedOperationException();
   }
+
+  /**
+   * Delete records up to (and include) the corresponding offset for all system stream partitions specified in the map
```

Comment:
I think 'trim' is pretty descriptive of what this is doing, so wouldn't mind calling it that.

Else, 'records' is a Kafka specific term. Samza refers to these as simply messages, so maybe deleteMessages? Or clearUpTo(Map<SSP, String> offset)?

Minor: "Delete records up to (and **including**) the **provided ssp offsets**."

Reply:
Thanks @prateekm for your review! I have updated the code and the comment as suggested.

I think a few words are very similar, e.g. clear, delete, trim, purge. I don't have a strong preference among them as long as we use the name consistently. I used the name `deleteMessages` for now. I will ask you in person later regarding the name.